feature-group-definitions/gradients.yml

```diff
@@ -0,0 +1,50 @@
+spec: https://drafts.csswg.org/css-images-4/#gradients
+caniuse:
+  - css-gradients
+  - css-repeating-gradients
+  - css-conic-gradients
+compat_features:
+  # Background-image prop
+  - css.properties.background-image.gradients
+  - css.properties.background-image.multiple_backgrounds
+  # CSS gradient image type
+  - css.types.image.gradient
```

Comment:
So far, we haven't rolled up everything required to use a feature, only the distinctive things that make up that feature. So I think we can omit `background-image` and the `gradient` type.

See also: https://github.com/web-platform-dx/web-features/issues/217

```suggestion
```

Reply:
Ok, I think that makes sense, but I can see cases when this could be a problem, in theory.

Imagine a new CSS value `foo`. Let's say this value can be used by `bar` and `baz`.
What if one browser supports both `bar:foo` and `baz:foo` but another browser only supports `bar:foo`.

Maybe `baz:foo` is a cool feature that we want to document. If we don't add `baz` to the list of compat features, then we don't know if browser really support it.

Reply:
> Imagine a new CSS value `foo`. Let's say this value can be used by `bar` and `baz`. What if one browser supports both `bar:foo` and `baz:foo` but another browser only supports `bar:foo`.
> 
> Maybe `baz:foo` is a cool feature that we want to document. If we don't add `baz` to the list of compat features, then we don't know if browser really support it.

Yes, I agree that this is a real concern: declaring support for a type, abstractly, doesn't say much about the practical outcome of using it with a specific property. That said, do you have a feeling about what properties should and should not qualify for this group?

I have a concrete example here: `background-image: <gradient>` works and so it doesn't alter the overall support story here. But what about `content: <gradient>`? It's specified as accepting a `<gradient>` value, but [the support story is not so good](https://developer.mozilla.org/en-US/docs/Web/CSS/content#browser_compatibility). Should we say that the overall CSS gradients feature isn't supported until that lands too? If not, what's the criteria for choosing some properties and not others?

I genuinely do not know the answer to these questions, so I'm more than open to ideas here.

---

Some optional background about BCD:

CSS types are probably BCD's weakest data. I've never felt good about what it covers. I believe the (undocumented) convention is to report a type as supported when the type is first supported as the value of _any_ property. That's probably good enough for the case of "is this type supported in any way?" but probably not so much for "can I use this type with property `baz`?

In a better world, BCD would also (or only) have keys along the lines of `css.properties.background-image.gradient.{conic-gradient,linear-gradient,…}`. Then we could say with some certainty whether you could _use_ a given CSS type across the properties where it's an allowed value.

Reply:
Hi @captainbrosset, did you have any thoughts on this question?

> what properties should and should not qualify for this group?

(otherwise we can plow ahead with this and wait for the question to come up again)

Reply:
> I have a concrete example here: background-image: <gradient> works and so it doesn't alter the overall support story here. But what about content: <gradient>? It's specified as accepting a <gradient> value, but [the support story is not so good](https://developer.mozilla.org/en-US/docs/Web/CSS/content#browser_compatibility). Should we say that the overall CSS gradients feature isn't supported until that lands too? If not, what's the criteria for choosing some properties and not others?

We probably shouldn't gate CSS gradients on the fact that it's suported for `content`. So I'm fine with removing the lines you suggested.

It does, however, make it harder to reason about what goes into a group and what doesn't.

In that particular example, where would we signify that gradients aren't support in `content` yet? Would we make another group (maybe named `full-gradient`, or `gradient-v2-final`) that contains this group + the support for `content`?

It seems this project should be able to say if support for gradient in `content` is here or not. Do you agree?
If so, then I think people would naturally go to the `gradient` group to find this information.

Maybe that means that support for a group isn't only the "sum of its parts". Maybe some of the listed parts shouldn't count towards the group's overall suppor, but still be here for decision making. In this case, gradient for `content` would be listed somehow, maybe as a note, but wouldn't make the group become red.

Reply:
So I just searched for "css gradients" in the hopes of getting a better idea of what web developers are hearing about when they hear about "gradients." I looked at the first several search results and also checked the docs for Tailwind and Bootstrap. With the major exception of MDN, everybody talks about _background_ gradients.

<details>
<summary>Some sample reading</summary>

- [CSS Gradient — Generator, Maker, and Background](https://cssgradient.io/) - Top result on Google, generates `background` properties. Several other generators in the top results do the same and caniuse links to [Ultimate CSS Gradient Generator - ColorZilla.com](https://www.colorzilla.com/gradient-editor/).
- [Gradient Color Stops - Tailwind CSS](https://tailwindcss.com/docs/gradient-color-stops) - In the _Backgrounds_ section
- [CSS Gradients - W3Schools](https://www.w3schools.com/css/css3_gradients.asp) - Examples all `background-image`
- [CSS Gradients | CSS-Tricks - CSS-Tricks](https://css-tricks.com/css3-gradients/) - Lede: "Just as you can declare the background of an element to be a solid color in CSS, you can also declare that background to be a gradient."
- [Make Beautiful Gradients in CSS, with linear-gradient, radial-gradient, or conic-gradient.](https://www.joshwcomeau.com/css/make-beautiful-gradients/) - Josh W. Comeau doesn't say it explicitly, but the examples use `background` and his generator uses `background-image`
- [Background · Bootstrap v5.3](https://getbootstrap.com/docs/5.3/utilities/background/#background-gradient) - Again `background-image` here

</details>

To me this suggests a fairly straightforward way out: we rename this feature to be `background-gradients`. This is consistent with commonly-used terminology and makes the scope of the feature explicit. Should we decide to cover `content` gradients, then we have a natural follow-up: `content-gradients`. We also leave the door open to some kind omnibus group, like `gradients` (containing both `content-gradients` and `background-gradients`).

(As a bonus, it also makes it less likely for us to collide with SVG gradients.)